### DISCUSSION TITLE
Currently getting loads of issues with metric Fargate

### DISCUSSION THREAD

**Discussion by rbtaina** [artilleryio/artillery#3426] [Artillery]
- Error: error sending test data to Artillery Cloud
Test report may be incomplete
Warning: multiple batches of metrics for period 1733832590000

This only started when i add   includeFiles: to my yml file

Any recommendations?
Thanks Robert

**Comment by hassy** [comment]
Hi @rbtaina - hard to tell what's going on without more information. There seem to be two distinct issues here:

1. Error with sending data to Artillery Cloud - is this happening consistently or only for some tests? Do you see that error immediately when a test starts, or only after some time? Do you see the test report for that test in Artillery Cloud?
2. "Multiple batches of metrics" can indicate an issue, but not always.

**Reply by rbtaina** [reply]
Hey @hassy sorry for not adding more information as i was still trying to understand the issue. 

It seems that this happens when i have tests running, and due to slow connection or maximizing the box memory and cpu, the test might not finish, therefore the test will just sit and wait in the background to be terminated, when running through cloud i can`t terminate that session, which will cause the issues above, however if i encounter the same when running locally and running in headless if i manually close that session the tests will just carry on executing.

Do you have anything or recommend anything that could help with closing tests where the locator has been missed or terminate a test etc? what has helped was using 
const sleep = (delay) => new Promise((resolve) => setTimeout(resolve, delay))
    await sleep(1000)

It has helped with my tests but not the ideal fix.

Hope the above helps.
Robert

**Reply by hassy** [reply]
Artillery should clean up automatically when there's a failure like that (missing locator etc). It does sound like the Fargate workers are getting overloaded, so everything slows down to a crawl. We published a more detailed guide for load testing with Playwright recently here: https://www.artillery.io/docs/playwright As a rule of thumb we'd recommend at least 1 vCPU per concurrent virtual user. You can increase the number of CPUs per worker on Fargate up to 16 with the [`--cpu` flag](https://www.artillery.io/docs/reference/cli/run-fargate#:~:text=artilleryio/%3Csecret%2Dname%3E-,%2D%2Dcpu,-Added%20in). If that alone does not help, you'll need to reduce the number of VUs per worker (reduce `arrivalRate` values in the config) and increase the number of workers instead.
